src/DataShare/DataShare/Account/RemoveAzDataShareAccount.cs

```diff
@@ -0,0 +1,133 @@
+    using Microsoft.Azure.Commands.DataShare.Common;
+    using Microsoft.Azure.Commands.DataShare.Helpers;
+    using Microsoft.Azure.Management.DataShare;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Properties;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+
+    /// <summary>
+    /// Defines Remove-AzDataShareAccount cmdlets.
+    /// </summary>
+    [Cmdlet(
+         "Remove",
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShareAccount",
+         DefaultParameterSetName = ParameterSetNames.FieldsParameterSet,
+         SupportsShouldProcess = true),
+     OutputType(typeof(bool))]
+    public class RemoveAzDataShareAccount : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "The resource group name of azure data share account.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "Azure data share account name.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The resourceId of the azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource id of the azure data share account.",
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ResourceIdCompleter(ResourceTypes.Account)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Azure data share account object.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParameterSetNames.ObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "The azure data share account object.")]
+        [ValidateNotNullOrEmpty]
+        public PSDataShareAccount InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Return object (if specified).")]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter]
+        public SwitchParameter Force { get; set; }
```

Comment:
@gaurangisaxena the `-Force` parameter should be used in the case that we want to prompt the user for additional confirmation when performing an action. For example, if the user is deleting a resource that contains sub-resources that will also be deleted, then we should prompt the user to confirm that they also want to delete these sub-resources. Having the `-Force` parameter will allow the user to bypass these prompts and continue execution.

If your cmdlet doesn't need this additional prompt, then we should remove the `-Force` parameter.

Reply:
We do have sub resources for most of the resource types, especially account. Should we get rid of -Force in delete implementation of all the resource types?

Reply:
If the deletion will remove the sub-resources as well, then it should be fine to keep the `-Force` parameter.

Reply:
Yes it removes sub resources as well. Keeping -Force parameter